README.md

```diff
@@ -20,11 +20,11 @@ Argo is a [Cloud Native Computing Foundation (CNCF)](https://cncf.io/) graduated
 
 ## Use Cases
 
-* [Machine Learning pipelines](https://argoproj.github.io/argo-workflows/use-cases/machine-learning/)
-* [Data and batch processing](https://argoproj.github.io/argo-workflows/use-cases/data-processing/)
-* [Infrastructure automation](https://argoproj.github.io/argo-workflows/use-cases/infrastructure-automation/)
-* [CI/CD](https://argoproj.github.io/argo-workflows/use-cases/ci-cd/)
-* [Other use cases](https://argoproj.github.io/argo-workflows/use-cases/other/)
+* [Machine Learning pipelines](https://argo-workflows.readthedocs.io/en/v3.5.2/use-cases/machine-learning/)
```

Comment:
Ah I see, these changes will impact people who navigate to the `release-3.5` branch directly for the examples and docs.

The source code changes would only affect the next version of the binary though, which would be v3.5.3 at that time. Hmmm... using `release-3.5` or `3.5.x` for the version number in the docs permalink may make the most sense then